plugins/package-managers/cocoapods/src/funTest/kotlin/CocoaPodsFunTest.kt

```diff
@@ -74,5 +78,29 @@ class CocoaPodsFunTest : WordSpec({
 
             result.withInvariantIssues().toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
         }
+
+        "determine dependencies from a Podfile located in a subdirectory referring to npm packages" {
+            // The NPM package manager is run first to initialize the node_modules directory, which is required to
+            // parse the Podfile successfully.
+
+            val npmDefinitionFile = getAssetFile("projects/synthetic/subdirectory/package.json")
+            val result = analyze(
+                npmDefinitionFile.parentFile,
+                true,
+                packageManagers = listOf(NpmFactory(), CocoaPodsFactory()),
```

Comment:
Is NPM always the package manager used in this scenario, or could it be other node mgr?

Reply:
No any node manager work. Our customer uses Yarn.

I wrote the test with NPM for simplicity.